src/smw/ai.cpp

```diff
@@ -495,7 +507,7 @@ void CPlayerAI::Think(COutputControl * playerKeys)
 
             goto ExitDeathCheck;
         } else if ((ttLeftTile & tile_flag_solid) || (ttLeftTile & tile_flag_solid_on_top) || g_map->block(iDeathX1, iDeathY) ||
-                  ((ttRightTile & tile_flag_solid) && (ttRightTile & tile_flag_solid_on_top)) || g_map->block(iDeathX2, iDeathY)) {
+                  (ttRightTile & tile_flag_solid) || (ttRightTile & tile_flag_solid_on_top) || g_map->block(iDeathX2, iDeathY)) {
```

Comment:
What exactly is the effect of this change?

Reply:
That line is supposed to exit the death check with iFallDanger = 0, meaning there is no danger if there is a solid, solidtop, or block below your left or right side, and it got refactored wrong at some point.